validate.go

```diff
@@ -0,0 +1,56 @@
+package analytics
+
+func getString(msg map[string]interface{}, field string) string {
+	val, _ := msg[field].(string)
+	return val
+}
+
+func ValidateFields(msg map[string]interface{}) error {
```

Comment:
Enforcing a `map[string]interface{}` may be a bit too opinionated on the context in which this function would be used.

Have you considered other approaches?

Reply:
We could go back to the interface that requests the underlying field values, ie:

```
type FieldGetter interface {
    func GetField(field string) interface{}
}
```

And use that as the type to `ValidateFields`. 

What other use cases do you have in mind?